syntax = "proto2";

enum ECsgoGCMsg {
	k_EMsgGCCStrike15_v2_Base = 9100;
	k_EMsgGCCStrike15_v2_MatchmakingStart = 9101;
	k_EMsgGCCStrike15_v2_MatchmakingStop = 9102;
	k_EMsgGCCStrike15_v2_MatchmakingClient2ServerPing = 9103;
	k_EMsgGCCStrike15_v2_MatchmakingGC2ClientUpdate = 9104;
	k_EMsgGCCStrike15_v2_MatchmakingServerReservationResponse = 9106;
	k_EMsgGCCStrike15_v2_MatchmakingGC2ClientReserve = 9107;
	k_EMsgGCCStrike15_v2_MatchmakingClient2GCHello = 9109;
	k_EMsgGCCStrike15_v2_MatchmakingGC2ClientHello = 9110;
	k_EMsgGCCStrike15_v2_MatchmakingGC2ClientAbandon = 9112;
	k_EMsgGCCStrike15_v2_MatchmakingGCOperationalStats = 9115;
	k_EMsgGCCStrike15_v2_MatchmakingOperator2GCBlogUpdate = 9117;
	k_EMsgGCCStrike15_v2_ServerNotificationForUserPenalty = 9118;
	k_EMsgGCCStrike15_v2_ClientReportPlayer = 9119;
	k_EMsgGCCStrike15_v2_ClientReportServer = 9120;
	k_EMsgGCCStrike15_v2_ClientCommendPlayer = 9121;
	k_EMsgGCCStrike15_v2_ClientReportResponse = 9122;
	k_EMsgGCCStrike15_v2_ClientCommendPlayerQuery = 9123;
	k_EMsgGCCStrike15_v2_ClientCommendPlayerQueryResponse = 9124;
	k_EMsgGCCStrike15_v2_WatchInfoUsers = 9126;
	k_EMsgGCCStrike15_v2_ClientRequestPlayersProfile = 9127;
	k_EMsgGCCStrike15_v2_PlayersProfile = 9128;
	k_EMsgGCCStrike15_v2_PlayerOverwatchCaseUpdate = 9131;
	k_EMsgGCCStrike15_v2_PlayerOverwatchCaseAssignment = 9132;
	k_EMsgGCCStrike15_v2_PlayerOverwatchCaseStatus = 9133;
	k_EMsgGCCStrike15_v2_GC2ClientTextMsg = 9134;
	k_EMsgGCCStrike15_v2_Client2GCTextMsg = 9135;
	k_EMsgGCCStrike15_v2_MatchEndRunRewardDrops = 9136;
	k_EMsgGCCStrike15_v2_MatchEndRewardDropsNotification = 9137;
	k_EMsgGCCStrike15_v2_ClientRequestWatchInfoFriends2 = 9138;
	k_EMsgGCCStrike15_v2_MatchList = 9139;
	k_EMsgGCCStrike15_v2_MatchListRequestCurrentLiveGames = 9140;
	k_EMsgGCCStrike15_v2_MatchListRequestRecentUserGames = 9141;
	k_EMsgGCCStrike15_v2_GC2ServerReservationUpdate = 9142;
	k_EMsgGCCStrike15_v2_ClientVarValueNotificationInfo = 9144;
	k_EMsgGCCStrike15_v2_MatchListRequestTournamentGames = 9146;
	k_EMsgGCCStrike15_v2_MatchListRequestFullGameInfo = 9147;
	k_EMsgGCCStrike15_v2_GiftsLeaderboardRequest = 9148;
	k_EMsgGCCStrike15_v2_GiftsLeaderboardResponse = 9149;
	k_EMsgGCCStrike15_v2_ServerVarValueNotificationInfo = 9150;
	k_EMsgGCCStrike15_v2_ClientSubmitSurveyVote = 9152;
	k_EMsgGCCStrike15_v2_Server2GCClientValidate = 9153;
	k_EMsgGCCStrike15_v2_MatchListRequestLiveGameForUser = 9154;
	k_EMsgGCCStrike15_v2_Client2GCEconPreviewDataBlockRequest = 9156;
	k_EMsgGCCStrike15_v2_Client2GCEconPreviewDataBlockResponse = 9157;
	k_EMsgGCCStrike15_v2_AccountPrivacySettings = 9158;
	k_EMsgGCCStrike15_v2_SetMyActivityInfo = 9159;
	k_EMsgGCCStrike15_v2_MatchListRequestTournamentPredictions = 9160;
	k_EMsgGCCStrike15_v2_MatchListUploadTournamentPredictions = 9161;
	k_EMsgGCCStrike15_v2_DraftSummary = 9162;
	k_EMsgGCCStrike15_v2_ClientRequestJoinFriendData = 9163;
	k_EMsgGCCStrike15_v2_ClientRequestJoinServerData = 9164;
	k_EMsgGCCStrike15_v2_ClientRequestNewMission = 9165;
	k_EMsgGCCStrike15_v2_GC2ClientTournamentInfo = 9167;
	k_EMsgGC_GlobalGame_Subscribe = 9168;
	k_EMsgGC_GlobalGame_Unsubscribe = 9169;
	k_EMsgGC_GlobalGame_Play = 9170;
	k_EMsgGCCStrike15_v2_AcknowledgePenalty = 9171;
	k_EMsgGCCStrike15_v2_Client2GCRequestPrestigeCoin = 9172;
	k_EMsgGCCStrike15_v2_GC2ClientGlobalStats = 9173;
	k_EMsgGCCStrike15_v2_Client2GCStreamUnlock = 9174;
	k_EMsgGCCStrike15_v2_FantasyRequestClientData = 9175;
	k_EMsgGCCStrike15_v2_FantasyUpdateClientData = 9176;
	k_EMsgGCCStrike15_v2_GCToClientSteamdatagramTicket = 9177;
	k_EMsgGCCStrike15_v2_ClientToGCRequestTicket = 9178;
	k_EMsgGCCStrike15_v2_ClientToGCRequestElevate = 9179;
	k_EMsgGCCStrike15_v2_GlobalChat = 9180;
	k_EMsgGCCStrike15_v2_GlobalChat_Subscribe = 9181;
	k_EMsgGCCStrike15_v2_GlobalChat_Unsubscribe = 9182;
	k_EMsgGCCStrike15_v2_ClientAuthKeyCode = 9183;
	k_EMsgGCCStrike15_v2_GotvSyncPacket = 9184;
	k_EMsgGCCStrike15_v2_ClientPlayerDecalSign = 9185;
	k_EMsgGCCStrike15_v2_ClientLogonFatalError = 9187;
	k_EMsgGCCStrike15_v2_ClientPollState = 9188;
	k_EMsgGCCStrike15_v2_Party_Register = 9189;
	k_EMsgGCCStrike15_v2_Party_Unregister = 9190;
	k_EMsgGCCStrike15_v2_Party_Search = 9191;
	k_EMsgGCCStrike15_v2_Party_Invite = 9192;
	k_EMsgGCCStrike15_v2_Account_RequestCoPlays = 9193;
	k_EMsgGCCStrike15_v2_ClientGCRankUpdate = 9194;
	k_EMsgGCCStrike15_v2_ClientRequestOffers = 9195;
	k_EMsgGCCStrike15_v2_ClientAccountBalance = 9196;
	k_EMsgGCCStrike15_v2_ClientPartyJoinRelay = 9197;
	k_EMsgGCCStrike15_v2_ClientPartyWarning = 9198;
	k_EMsgGCCStrike15_v2_SetEventFavorite = 9200;
	k_EMsgGCCStrike15_v2_GetEventFavorites_Request = 9201;
	k_EMsgGCCStrike15_v2_ClientPerfReport = 9202;
	k_EMsgGCCStrike15_v2_GetEventFavorites_Response = 9203;
	k_EMsgGCCStrike15_v2_ClientRequestSouvenir = 9204;
	k_EMsgGCCStrike15_v2_ClientReportValidation = 9205;
	k_EMsgGCCStrike15_v2_GC2ClientRefuseSecureMode = 9206;
	k_EMsgGCCStrike15_v2_GC2ClientRequestValidation = 9207;
	k_EMsgGCCStrike15_v2_ClientRedeemMissionReward = 9209;
	k_EMsgGCCStrike15_ClientDeepStats = 9210;
	k_EMsgGCCStrike15_StartAgreementSessionInGame = 9211;
}

message CMsgGCCStrike15_v2_Party_Search {
	optional uint32 ver = 1;
	optional uint32 apr = 2;
	optional uint32 ark = 3;
	repeated uint32 grps = 4;
	optional uint32 launcher = 5;
	optional uint32 game_type = 6;
}

message CMsgGCCStrike15_v2_Party_SearchResults {
	message Entry {
		optional uint32 id = 1;
		optional uint32 grp = 2;
		optional uint32 game_type = 3;
		optional uint32 apr = 4;
		optional uint32 ark = 5;
		optional uint32 loc = 6;
	}

	repeated .CMsgGCCStrike15_v2_Party_SearchResults.Entry entries = 1;
}

message CMsgGCCStrike15_v2_Party_Invite {
	optional uint32 accountid = 1;
	optional uint32 lobbyid = 2;
}

message DetailedSearchStatistic {
	optional uint32 game_type = 1;
	optional uint32 search_time_avg = 2;
	optional uint32 players_searching = 4;
}

message GlobalStatistics {
	optional uint32 players_online = 1;
	optional uint32 servers_online = 2;
	optional uint32 players_searching = 3;
	optional uint32 servers_available = 4;
	optional uint32 ongoing_matches = 5;
	optional uint32 search_time_avg = 6;
	repeated .DetailedSearchStatistic search_statistics = 7;
	optional string main_post_url = 8;
	optional uint32 required_appid_version = 9;
	optional uint32 pricesheet_version = 10;
	optional uint32 twitch_streams_version = 11;
	optional uint32 active_tournament_eventid = 12;
	optional uint32 active_survey_id = 13;
	optional uint32 rtime32_cur = 14;
	optional uint32 rtime32_event_start = 15;
}

message IpAddressMask {
	optional uint32 a = 1;
	optional uint32 b = 2;
	optional uint32 c = 3;
	optional uint32 d = 4;
	optional uint32 bits = 5;
	optional uint32 token = 6;
}

message CMsgGCCStrike15_v2_MatchmakingGC2ClientUpdate {
	message Note {
		optional int32 type = 1;
		optional int32 region_id = 2;
		optional float region_r = 3;
		optional float distance = 4;
	}

	optional int32 matchmaking = 1;
	repeated uint32 waiting_account_id_sessions = 2;
	optional string error = 3;
	repeated uint32 ongoingmatch_account_id_sessions = 6;
	optional .GlobalStatistics global_stats = 7;
	repeated uint32 failping_account_id_sessions = 8;
	repeated uint32 penalty_account_id_sessions = 9;
	repeated uint32 failready_account_id_sessions = 10;
	repeated uint32 vacbanned_account_id_sessions = 11;
	optional .IpAddressMask server_ipaddress_mask = 12;
	repeated .CMsgGCCStrike15_v2_MatchmakingGC2ClientUpdate.Note notes = 13;
	repeated uint32 penalty_account_id_sessions_green = 14;
	repeated uint32 insufficientlevel_sessions = 15;
	repeated uint32 vsncheck_account_id_sessions = 16;
	repeated uint32 launcher_mismatch_sessions = 17;
	repeated uint32 insecure_account_id_sessions = 18;
}